core/src/Helpers/SortedSetExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Identity.Core.Helpers
+{
+    internal static class SortedSetExtensions
+    {
+        public static bool ScopeContains(this SortedSet<string> scopes, SortedSet<string> otherScope)
```

Comment:
>SortedSet [](start = 46, length = 9)

There seem to be a lot of helpers around sorted sets and transformations to string. This is only used for handling scopes as far as I have seen. It might be better to group methods in a ScopeHelpers class rather than adding to "SortedSet" and "String". In particular, string is so wildly used that it would become inconvenient to see "AsLowerCaseSortedSet" and "AsArray" etc. every time IntelliSense is invoked on a string.

Reply:
Really good point.  I'll go through the sorted set and string extensions and see if I can consolidate them into scope specific methods.